Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputSystem.cs

```diff
@@ -115,41 +116,55 @@ public interface IMixedRealityInputSystem : IMixedRealityEventSystem
         /// Raise the event that the Input Source was detected.
         /// </summary>
         /// <param name="source">The detected Input Source.</param>
-        void RaiseSourceDetected(IMixedRealityInputSource source);
+        /// <param name="controller"></param>
+        void RaiseSourceDetected(IMixedRealityInputSource source, IMixedRealityController controller = null);
```

Comment:
Dont see how you can detect / lose a null controller

Reply:
For example, voice does not have a controller.